MRN-Code/coinstac
# Working Single Shot

# Problem
* Closes #417 
* Closes #414 
* Symlink directory doesn't match pipeline expectations
* Single-shot and multi-shot won't run in UI
# Solution
* Run files unlinked after results or error found on run
* Run errors display notifications and are saved along with local run object
* Update symlink directory to use clientId and runId
* Update comp specs for multi and single shot regression
* Update test-setup to include a single-shot pipeline
* Temporarily remove calls to automatically download images from Docker Hub 
# Testing
* Single-shot is not yet on Docker Hub.
   * Clone it: `git clone git@github.com:MRN-Code/ssr_test.git`
   * Change to branch with file parser: `git checkout parser-filepath-edit`
   * Build Docker image locally: `docker build -t ssr_test .`
* Start application, login as `test`.
* Add two local file groups:
   * Click on **Collections** nav item
   * Click on **Test Collection** details button
   * Click on **Files** tab
   * Click **Add Group**
   * Click the *metadata file* option and hit **Add Files Group**
   * Find the following file and click it to add `~/coinstac/packages/coinstac-ui/test/e2e/test_files/site1/site1_Covariate.csv`
   * Add a second file group by choosing the  *manually select files* option and hit **Add Files Group**
   * Select the first 21 text files (`subject0_aseg_stats.txt` - `subject20_aseg_stats.txt`) from the directory `~/coinstac/packages/coinstac-ui/test/e2e/test_files/site1`
      * This might take ~10 seconds due to the sorting (#396 )
* Map local groups to computation vars
   * Click on **Consortia** tab in the same collection
   * Choose *Text Consortia 2* under the **Select Consortia** dropdown
   * Select the `.csv` extension group for both *covariates*
   * Select the `.txt` extension group for the *data* input
* Run SSR Pipeline
   * Click on **Consortia** nav item
   * **Test Consortia 2** should have a *Start Pipeline* button enabled, click it
   * When the run starts, click the **Home** nav item to view the status of the run as it progresses
